onmt/train_single.py

```diff
@@ -51,14 +51,14 @@ def training_opt_postprocessing(opt, device_id):
     if opt.rnn_size != -1:
         opt.enc_rnn_size = opt.rnn_size
         opt.dec_rnn_size = opt.rnn_size
-        if opt.model_type == 'text' and opt.enc_rnn_size != opt.dec_rnn_size:
-            raise AssertionError("""We do not support different encoder and
-                                 decoder rnn sizes for translation now.""")
+        # removed the part for model_type != text...why would that matter?
+        assert opt.enc_rnn_size == opt.dec_rnn_size, \
```

Comment:
@da03 this was you, can you have different enc_rnn_size and dec_rnn_size in the case of audio ?

Reply:
I had a look at `AudioEncoder` and it appears to include a bridge layer that allows for different-sized encoder and decoder rnns. I think that in the long run this is not how it should be because 1) A bridge between different-sized encoders and decoders is not specific to audio and 2) the audio encoder should not need to know what the decoder's dimensions are.

But for now, I'll add the bit about `model_type` back into the check.